chain/messagepool/messagepool.go

```diff
@@ -139,8 +139,24 @@ type MessagePool struct {
 
 	keyCache *lru.Cache[address.Address, address.Address]
 
-	curTsLk sync.RWMutex // DO NOT LOCK INSIDE lk
-	curTs   *types.TipSet
+	// curTs stands for 'current TipSet'
+	curTs *types.TipSet
+
+	// The transactionLk lock is used to ensure consistency when transacting between the TipSet (TS) and the Message Pool.
+	// Any modification to curTs must be wrapped in a transactionLk.
+	// This lock is not needed if you're only interested in a consistent view of the state without changing it.
+	transactionLk sync.RWMutex
```

Comment:
I believe this lock can be changed in to a `sync.Mutex` since the `RLock()` is never used?

Reply:
thank you! resolved and deleted